momentumopt/demos/demo_biped_from_python.py

```diff
@@ -0,0 +1,179 @@
+import pymomentum as mopt
+from pymomentum import PlannerSetting, ContactPlanFromFile,ContactState, DynamicsOptimizer, DynamicsState, ContactType, EffId, KinematicsSequence
+from pysolver import ExitCode
+import pathlib
+from math import floor
+import numpy as np
+import matplotlib.pyplot as plt
+
+"""
+This script show how to define a problem for the dynamicOptimizer from python.
+From a manually defined sequence of contact (with duration) it show how to optimize the centroidal dynamics 
+and retrieve the centroidal trajectories.
+It use a biped robot, walking 75cm straight with 6 steps
+"""
+
+PATH = pathlib.Path(__file__).absolute()
+PATH = PATH.parent.parent / 'config'
+print("PATH to config folder : ", PATH)
+
+# load default settings:
+planner_setting = PlannerSetting()
+planner_setting.initialize(str(PATH / 'cfg_biped.yaml'))
+# define the duration of the motion in python:
+duration = 11.4
+dt = planner_setting.get(mopt.PlannerDoubleParam_TimeStep)
+n_time_steps = int(floor(duration/dt))
+# update this values in planner_setting:
+planner_setting.set(mopt.PlannerDoubleParam_TimeHorizon, duration)
+planner_setting.set(mopt.PlannerIntParam_NumTimesteps, n_time_steps)
+
+# initialize an empty contact plan:
+contact_planner = ContactPlanFromFile()
+contact_planner.initialize(planner_setting)
+mopt_cs = contact_planner.contactSequence()
+
+# Add contact states corresponding to the biped walk from python:
+# We use a double support duration of 0.2second and single support of 1.4 seconds and a first double support phase of 1s
+
+# Create the contact sequence for the right foot:
+RF_position = np.array([0., -0.085, 0.])
+mopt_cs_RF = mopt_cs.contact_states(EffId.right_foot.value())
+cp = ContactState()
+cp.start_time = 0.
+cp.end_time = 1.
+cp.contactType = ContactType.FlatContact
+cp.active = True
+cp.position = RF_position
```

Comment:
If you want, the code might look easier if you create a helper function `createContactState(start_time, end_time, contact_type, active, position)`, which takes the arguments and puts them on the ContactState. This will change the current code to many one-liners per ContactState and thereby make it easier to read the code.

Reply:
Thanks for the suggestion, it is indeed easier to read.